scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -26,7 +26,7 @@ def _get_proxy(self, url, orig_type):
         proxy_url = urlunparse((proxy_type or orig_type, hostport, '', '', '', ''))
 
         if user:
-            user_pass = '%s:%s' % (unquote(user), unquote(password))
+            user_pass = to_bytes('%s:%s' % (unquote(user), unquote(password)))
```

Comment:
Yeah, there are general rules for http header encoding ("nothing works"), but some headers have specific encodings which are more common - e.g. cookies are commonly encoded to utf8 and Autorization headers are commonly encoded to latin1.